### PR TITLE
homeworks-for-week1-noodejs

### PR DESCRIPTION
**Pull request by masumesani**

### CODE REVIEW

File: nodejs/week1/warmup.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Circle{
+    constructor(radius){
+        this.radius = radius;
+    }
+
+    getDiameter= () => this.radius * 2;
+    getCircumference = () => 2 * Math.PI * this.radius;
+    getArea = () =>  Math.PI *(this.radius * this.radius);
```
Comment (orhantoy) [inline nodejs/week1/warmup.js]: Using the short-hand arrow syntax only seems to work from version 12 of Node. This is not a problem, just something to be aware of.

File: nodejs/week1/warmup.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+const circle = new Circle(10);
+circle.radius
+let test = circle.getDiameter();
+console.log(test)
+test = circle.getCircumference();
+console.log(test)
+
+
```
Comment (orhantoy) [inline nodejs/week1/warmup.js]: Remember to clean up these extra, empty lines.

File: nodejs/week1/mealSharingApp/src/backend/routes/cheap-meals.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = (req,res)=>{
+    
+    let meals = dataHelper.getMeals();
+    
+    if(!req.query.margin){
+
+        res.send(dataHelper.getMealsAugmentedWithReviews(meals));
+        
+    } else {
+        let margin = parseInt(req.query.margin);
```
Comment (orhantoy) [inline nodejs/week1/mealSharingApp/src/backend/routes/cheap-meals.js]: Very nice use of query parameters 💪

File: nodejs/week1/mealSharingApp/src/backend/routes/meal.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+    const meals = dataHelper.getMeals();
+
+    const randomMealIndex =  Math.floor(Math.random()* meals.length);
+    
+    const meal = meals[randomMealIndex];
+    
+    const result = dataHelper.getMealsAugmentedWithReviews([meal])
+
+    res.send(result);
```
Comment (orhantoy) [inline nodejs/week1/mealSharingApp/src/backend/routes/meal.js]: This route returns a JSON array with a single meal object, so I would probably change this to return the object only.

File: nodejs/week1/warmup.js (hunk 1 above)
Comment (masumesani) [reply nodejs/week1/warmup.js]: noted, thanks

File: nodejs/week1/mealSharingApp/src/backend/routes/meal.js (hunk 4 above)
Comment (masumesani) [reply nodejs/week1/mealSharingApp/src/backend/routes/meal.js]: done it.

File: nodejs/week1/warmup.js (hunk 2 above)
Comment (masumesani) [reply nodejs/week1/warmup.js]: wipe and clean

**Pull request merged**